pkg/producer/producer.go

```diff
@@ -0,0 +1,33 @@
+package producer
+
+import (
+	"fmt"
+	"github.com/aquasecurity/tracee/pkg/config"
+	"github.com/aquasecurity/tracee/pkg/errfmt"
+	"github.com/aquasecurity/tracee/types/trace"
+)
+
+// EventsProducer is a type that is able to generate events
+type EventsProducer interface {
```

Comment:
I think the eBPF event decoder could be considered such an object no? This could be a useful abstraction.

Reply:
Yea, this was the idea :)
I tried to fit to the current design